internal/controller/spinapp_controller.go

```diff
@@ -264,13 +264,13 @@ func (r *SpinAppReconciler) reconcileDeployment(ctx context.Context, app *spinv1
 
 		err = r.Client.Create(ctx, secret)
 		if err != nil {
-			if client.IgnoreAlreadyExists(err) == nil {
-				log.Debug("RuntimeConfig Secret already exists", "runtime_config_secret_name", secret.ObjectMeta.Name)
+			if client.IgnoreAlreadyExists(err) != nil {
+				return fmt.Errorf("failed to create RuntimeConfig secret: %w", err)
 			}
-			return fmt.Errorf("failed to create RuntimeConfig secret: %w", err)
+			log.Debug("RuntimeConfig Secret already exists", "runtime_config_secret_name", secret.ObjectMeta.Name)
+		} else {
+			log.Info("created runtimeconfig secret")
 		}
```

Comment:
```suggestion
		}
```

we can drop this log